In [1]:
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout, Activation, Flatten, Conv2D, MaxPooling2D
from tensorflow.keras import regularizers
from tensorflow.keras.utils import plot_model
import pickle
import numpy as np

In [2]:
X_train=pickle.load(open("X_train_descriptors","rb"))
y_train=pickle.load(open("y_train_labels","rb"))
X_test=pickle.load(open("X_test_descriptors","rb"))
y_test=pickle.load(open("y_test_labels","rb"))

In [3]:
X_train.shape

(297335, 128)

In [4]:
y_train.shape

(297335,)

In [21]:
def create_model():
    model = Sequential()
    model.add(Dense(1024, activation='relu', input_shape=(X_train.shape[1],)))
    #model.add(Dropout(0.2))
    model.add(Dense(512, activation='relu'))
    model.add(Dense(512, activation='relu'))
    model.add(Dense(512, activation='relu'))
    #model.add(Dropout(0.2))
    model.add(Dense(1024, activation='relu'))
    model.add(Dropout(0.2))
    model.add(Dense(1, activation='sigmoid'))
    model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
    return model

In [22]:
model=create_model()

In [23]:
model.fit(X_train,y_train,batch_size=64,epochs=30,validation_data=(X_test,y_test))

Epoch 1/30
4646/4646 [==============================] - 9s 2ms/step - loss: 0.6906 - accuracy: 0.5325 - val_loss: 0.6907 - val_accuracy: 0.5268
Epoch 2/30
4646/4646 [==============================] - 8s 2ms/step - loss: 0.6879 - accuracy: 0.5441 - val_loss: 0.6874 - val_accuracy: 0.5449
Epoch 3/30
4646/4646 [==============================] - 8s 2ms/step - loss: 0.6858 - accuracy: 0.5498 - val_loss: 0.6911 - val_accuracy: 0.5364
Epoch 4/30
4646/4646 [==============================] - 8s 2ms/step - loss: 0.6837 - accuracy: 0.5552 - val_loss: 0.6874 - val_accuracy: 0.5409
Epoch 5/30
4646/4646 [==============================] - 8s 2ms/step - loss: 0.6817 - accuracy: 0.5593 - val_loss: 0.6849 - val_accuracy: 0.5481
Epoch 6/30
4646/4646 [==============================] - 8s 2ms/step - loss: 0.6794 - accuracy: 0.5648 - val_loss: 0.6852 - val_accuracy: 0.5496
Epoch 7/30
4646/4646 [==============================] - 9s 2ms/step - loss: 0.6771 - accuracy: 0.5695 - val_loss: 0.6899 - val_accuracy:

After many times tries of change network, the accuracy does not reach 55%.
